### ISSUE TITLE
AttributeInferenceError: '__slots__' not found on ClassDef node

### ISSUE THREAD

**Issue by correctmost** [pylint-dev/astroid#3067]
### Steps to reproduce

The following code triggers an `AttributeInferenceError` exception that is not caught by Pylint:

```python
class C:
    __slots__: None

    def m(self):
        self.a = 1
```

I found the bug during a local fuzzing run.

### Current behavior
```python-traceback
Traceback (most recent call last):
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2450, in igetattr
    attributes = self.getattr(name, context, class_context=class_context)
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2383, in getattr
    raise AttributeInferenceError(target=self, attribute=name, context=context)
astroid.exceptions.AttributeInferenceError: '__slots__' not found on <ClassDef.C l.1 at 0x7f8063c70500>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "pylint/pylint/utils/ast_walker.py", line 87, in walk
    callback(astroid)
    ~~~~~~~~^^^^^^^^^
  File "pylint/pylint/checkers/classes/class_checker.py", line 1724, in visit_assignattr
    self._check_in_slots(node)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "pylint/pylint/checkers/classes/class_checker.py", line 1782, in _check_in_slots
    slots = klass.slots()
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2813, in slots
    return self._all_slots
           ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.14/functools.py", line 1126, in __get__
    val = self.func(instance)
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2806, in _all_slots
    slots = list(grouped_slots(mro))
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2791, in grouped_slots
    cls_slots = cls._slots()
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2763, in _slots
    first = next(slots)
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2711, in _islots
    for slots in self.igetattr("__slots__"):
                 ~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "astroid/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2524, in igetattr
    raise InferenceError(
        str(error), target=self, attribute=name, context=context
    ) from error
astroid.exceptions.InferenceError: '__slots__' not found on <ClassDef.C l.1 at 0x7f8063c70500>.
```

### Expected behavior
No crash

### Version
649ec7d37d182
